site/src/pages/WorkspacePage/WorkspaceTopbar.stories.tsx

```diff
@@ -73,6 +96,86 @@ export const WithDeadline: Story = {
   },
 };
 
+const in30Minutes = new Date();
+in30Minutes.setMinutes(in30Minutes.getMinutes() + 30);
+export const WithApproachingDeadline: Story = {
+  args: {
+    workspace: {
+      ...MockWorkspace,
+      latest_build: {
+        ...MockWorkspace.latest_build,
+        deadline: in30Minutes.toISOString(),
+      },
+    },
+  },
+  play: async ({ canvasElement, step }) => {
+    const canvas = within(canvasElement);
+
+    await step("activate hover trigger", async () => {
+      await userEvent.hover(canvas.getByTestId("schedule-controls-autostop"));
+      await waitFor(() =>
+        expect(screen.getByRole("tooltip")).toHaveTextContent(
+          /this workspace has enabled autostop/,
+        ),
+      );
+    });
+  },
+};
+
+const in8Hours = new Date();
+in8Hours.setHours(in8Hours.getHours() + 8);
+export const WithFarAwayDeadline: Story = {
+  args: {
+    workspace: {
+      ...MockWorkspace,
+      latest_build: {
+        ...MockWorkspace.latest_build,
+        deadline: in8Hours.toISOString(),
+      },
+    },
+  },
+  play: async ({ canvasElement, step }) => {
+    const canvas = within(canvasElement);
+
+    await step("activate hover trigger", async () => {
+      await userEvent.hover(canvas.getByTestId("schedule-controls-autostop"));
+      await waitFor(() =>
+        expect(screen.getByRole("tooltip")).toHaveTextContent(
+          /this workspace has enabled autostop/,
+        ),
+      );
+    });
+  },
+};
+
+export const WithFarAwayDeadlineRequiredByTemplate: Story = {
+  args: {
+    workspace: {
+      ...MockWorkspace,
+      latest_build: {
+        ...MockWorkspace.latest_build,
+        deadline: in8Hours.toISOString(),
+      },
+    },
+    template: {
+      ...MockTemplate,
+      allow_user_autostop: false,
+    },
+  },
+  play: async ({ canvasElement, step }) => {
+    const canvas = within(canvasElement);
+
+    await step("activate hover trigger", async () => {
+      await userEvent.hover(canvas.getByTestId("schedule-controls-autostop"));
+      await waitFor(() =>
+        expect(screen.getByRole("tooltip")).toHaveTextContent(
+          /template has an autostop requirement/,
+        ),
+      );
+    });
+  },
+};
```

Comment:
Is there a way to tell to Storybook to screenshot after certain action? Eg. Screenshot after hovering or clicking an element.

Reply:
if you define a `play` function, chromatic defaults to snapshotting after

Reply:
and if you want granularity, you just define multiple stories for it to snapshot

Reply:
Makes sense!